### PR TITLE
[GROUP] MRC-1090 placemark search

### PR DESCRIPTION
**Pull request by tarakreft**
- [x] This change is documented in CHANGELOG.md
- [x] rename showTagsControl to showSearch or allowSearch (and add demo)
- [x] mock new getAllPlacemarks API
- [x] test visibility of search icon/menu based on props to hide (like showTagsControl)
- [x] test loadPlacemarks (create and update)
- [x] test not grouping Placemarks By ID

### CODE REVIEW

File: src/MapComponent.tsx [hunk 1]
```diff
@@ -781,10 +783,10 @@ class MapComponent extends Component<MapComponentProps, MapComponentState> {
                     locationID={locationID}
                     floorID={floorID}
                     api={api}
-                    markers={placemarks}
+                    placemarkOptions={placemarks}
```
Comment (vercasson) [inline src/MapComponent.tsx:786]: `placemarks` here refers to the main constructor key "placemarks" which refers to the options (CreateMapPlacemarksOptions) for displaying/loading placemarks. Same for `tags` (CreateMapTagsOptions)

https://github.com/arubanetworks/meridian-web-sdk/blob/MRC-1090-placemarkSearch/src/web-sdk.tsx#L214

File: src/PlacemarkLayer.tsx [hunk 2]
```diff
@@ -18,13 +18,28 @@ export interface PlacemarkLayerProps {
   locationID: string;
   floorID: string;
   api: API;
-  markers: CreateMapOptions["placemarks"];
+  placemarkOptions: CreateMapOptions["placemarks"];
   onPlacemarkClick: (placemark: PlacemarkData) => void;
   onUpdate: MapComponentProps["onPlacemarksUpdate"];
-  placemarks: Record<string, PlacemarkData>;
```
Comment (vercasson) [inline src/PlacemarkLayer.tsx:24]: This component fetches its own data now.

File: src/PlacemarkLayer.tsx [hunk 3]
```diff
@@ -68,21 +111,23 @@ export default class PlacemarkLayer extends Component<PlacemarkLayerProps> {
   }
 
   render() {
-    const { markers, onPlacemarkClick, mapZoomFactor, selectedItem } =
-      this.props;
```
Comment (vercasson) [inline src/PlacemarkLayer.tsx:72]: I reference these below now with `this.props`. I did this at one point to help with refactoring as I had state and props with similar names.

File: demo/cypress/utils/fake-api.js [hunk 4]
```diff
@@ -73,7 +78,7 @@ class FakeAPI {
     floorID,
     onInitialTags,
     // onTagLeave,
-    onTagUpdate
+    onTagUpdate,
     // onException,
     // onClose
```
Comment (amyhliao) [inline demo/cypress/utils/fake-api.js:83]: Are these commented out code needed?

**Comment by tarakreft** [comment]
Looks good!

**Review by amyhliao — APPROVED** [review]

File: demo/cypress/utils/fake-api.js (hunk 4 above)
Comment (vercasson) [reply demo/cypress/utils/fake-api.js:83]: These are purposely commented out to expose API functionality that we might need in the future. We don't do this very often though, so good catch @amyhliao

**Pull request merged**